.github/workflows/deploy-sphinx-docs-to-ghpages.yml

```
@@ -0,0 +1,63 @@
+name: Deploy Sphinx documentation
+
+
+on:
+
+  workflow_call:
+    inputs:
+        python-version:
+          required: true
+          type: string
+        path-doc:
+          required: false
+          type: string 
+          default: ./doc
+       
+    
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  
+  build:
+    
+    runs-on: ubuntu-latest
+
+    
+    steps:
+      
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }} 
+      - name: Deploy documentation
+        run: |
+          pip install .[docs]
+          sudo apt install pandoc
+          cd ${{ inputs.path-doc }}
+          make html
```

You can split in two steps:
```suggestion
      - name: Install dependencies
        run: |
          pip install .[docs]
          sudo apt install pandoc
      - name: Compile documentation
        working-directory: ${{ inputs.path-doc }}
        run: |
          make html
```